e2e/playwright/tests/@gitops/test.spec.ts

```diff
@@ -0,0 +1,230 @@
+  await expect(page.getByText('Are you sure you want to disable GitOps for this application?')).toBeVisible();
+  await page.getByRole('button', { name: 'Disable GitOps' }).click();
+  await expect(page.getByText('Are you sure you want to disable GitOps for this application?')).not.toBeVisible();
+  await expect(page.getByText('Not Enabled')).toBeVisible();
+  console.log('gitops disabled')
+
+  // visit application page, ensure deploy and redeploy buttons are visible, deploy new version
+  await page.getByText('Application', { exact: true }).click();
+  await expect(page.getByRole('button', { name: 'Redeploy' })).toBeVisible();
+  await expect(page.getByRole('button', { name: 'Deploy', exact: true })).toBeVisible();
+  await page.waitForTimeout(1000);
+  await page.getByRole('button', { name: 'Deploy', exact: true }).click();
+  await expect(page.getByText('(Sequence ')).toBeVisible();
+  await page.getByRole('button', { name: 'Yes, Deploy' }).click();
+  await expect(page.getByText('Currently deployed version')).toBeVisible(); // ensure that a version is deployed
+  await expect(page.locator('#app')).toContainText('Ready', { timeout: 30000 });
+  console.log('new version deployed')
+
+  // test reenabling gitops but with a failed ssh connection
+  await page.locator('div').filter({ hasText: /^GitOps$/ }).click();
+  await page.getByPlaceholder('owner').click();
+  await page.getByPlaceholder('owner').fill(gitopsOwner);
+  await page.getByPlaceholder('Repository').click();
+  await page.getByPlaceholder('Repository').fill(gitopsRepo);
+  await page.getByPlaceholder('main').click();
+  await page.getByPlaceholder('main').fill(randomBranch);
+  await page.getByPlaceholder('/path/to-deployment').click();
+  await page.getByPlaceholder('/path/to-deployment').fill(randomPath);
+  await page.getByRole('button', { name: 'Generate SSH key' }).click();
+  await page.getByRole('button', { name: 'Test connection to repository' }).click();
+  await expect(page.getByText('Connection to repository failed')).toBeVisible(); // there is no key in the repo
+  await page.getByRole('button', { name: 'Try again' }).click();
+  await expect(page.getByText('Connection to repository failed')).toBeVisible();
+  await page.getByRole('button', { name: 'Cancel' }).click();
+  await page.getByRole('button', { name: 'Back to configuration' }).click();
+  await expect(page.getByText('Repository access needed')).toBeVisible(); // ensure that the 'there's no key in the repo' message is now visible
+  await page.getByText('Application', { exact: true }).click();
+  await expect(page.getByRole('button', { name: 'Redeploy' })).toBeVisible(); // ensure that we're still in non-gitops mode
+  console.log('test complete')
+});
+
+let isChecked = false;
+async function trivialConfig(page: Page, isGitops: boolean) {
+  await page.getByRole('list').getByRole('link', { name: 'Config' }).click();
+  await expect(page.getByText('A trivial config item')).toBeVisible();
+  if (isChecked) {
```

Comment:
why not just do this:
```suggestion
  if (page.getByLabel('Trivial Config').isChecked()) {
```